src/status_im/contexts/communities/events.cljs

```diff
@@ -223,16 +223,19 @@
 
 (defn toggle-share-all-addresses
   [{:keys [db]} [community-id]]
-  (let [share-all-addresses? (get-in db [:communities community-id :share-all-addresses?])
-        accounts             (utils/sorted-non-watch-only-accounts db)
-        addresses            (set (map :address accounts))]
+  (let [share-all-addresses?      (get-in db [:communities community-id :share-all-addresses?])
+        next-share-all-addresses? (not share-all-addresses?)
+        accounts                  (utils/sorted-non-watch-only-accounts db)
+        addresses                 (set (map :address accounts))]
     {:db (update-in db
                     [:communities community-id]
-                    (fn [community]
-                      (-> community
-                          (assoc :share-all-addresses? (not share-all-addresses?))
-                          (cond-> (not share-all-addresses?)
-                                  (assoc :selected-permission-addresses addresses)))))}))
+                    assoc
+                    :share-all-addresses?          next-share-all-addresses?
+                    :selected-permission-addresses addresses)
```

Comment:
Hi @ajayesivan,
wouldn't this always share all addresses when we are toggling? (even if we are disabling sharing)

Reply:
ok got, this is expected behavior. We keep shared all addresses, but only change if we are allowed to deselect or not, based on `share-all-addresses?` flag

Reply:
Disabling the toggle implies that the previous state was enabled, indicating that all addresses were selected. Consequently, in both situations, whether the toggle is enabled or disabled, the sharing of all addresses doesn't make any difference. Do you see any potential issues with this, or does it make sense to you?